test/internal/helpers/Player.java

```diff
@@ -51,7 +50,7 @@ public static void addItem(int itemId) {
   }
 
   public static void addItem(int itemId, int count) {
-    AdventureResult.addResultToList(KoLConstants.inventory, ItemPool.get(itemId));
+    AdventureResult.addResultToList(KoLConstants.inventory, ItemPool.get(itemId, count));
```

Comment:
Not introduced in this PR but I think this function and the other should probably both point to a third function that takes an `AdventureResult`

Reply:
Well, the other one takes a name and makes a tallyItem. That can be used for things that are not actually items. For example, "pirate insult" or "Arena flyer ML". Those (intentionally) end up with itemId = -1, but have ITEM_PRIORITY, so work well on lists of AdventureResults that hold items. Like KoLConstants.tally - the "results" list that is shown in the GUI. Or even KoLConstants.inventory, although the non-items I mentioned are not put there by the actual KoLmafia code that creates them.

Perhaps a better API would have addItem use ItemPool.get and add to inventory and addTallyItem create a tallyitem and add to tally.

The other one also generates Cleanups and this one does not. I suppose the Cleanups version could call the other one to add to the list.

Reply:
I think we want this one to also make cleanups. It was made before Cleanups existed. As far as I know, all usages use it for items - we don't have any testing that uses tallyitems

Reply:
OK. I'll refactor.

Reply:
When I created these I didn't (and still don't) really understand the difference between each of the different methods -- I was just trying to find one that worked.

`tallyItem` was used because I was previously using `AdventureResult.parseResult("baconstone")` but that gave the wrong result while `tallyItem` worked. If there's a better "this is an item, parse this exactly" method that should probably be used instead.

Reply:
new AdventureResult("baconstone") should work. But if yu want to guarantee you get an item, ItemDatabase.getItemId("baconstone", 1, false) will look up the name, assuming it is not a plural (count===1) without fuzzy matching (false).

OK. I am submitting an improved refactoring. The tests still pass, so I haven't broken anything, it seems.

Note that all variants addItem always returns Cleanups, so you can use it in try-with-resources - or do a bit of unnecessary allocation if you don't do that.

Reply:
By the way - perhaps we want ItemPool.get(String name, int count) which calls ItemDatabase, in order to make it very clear which function you should call to get an AdventureResult.